test/process-env-port.js

```diff
@@ -0,0 +1,41 @@
+var test = require('tap').test,
+    request = require('request'),
+    spawn = require('child_process').spawn,
+    insanePorts = [ -Infinity, 1023, 65537, Infinity, 'wow', null, undefined]
+
+test('sane port', function (t) {
+  t.plan(2)
+  process.env.PORT = getRandomInt(1025, 65536)
+  var ecstatic = spawn(process.execPath, [__dirname + '/../lib/ecstatic.js'])
+  ecstatic.stdout.on('data', function (data) {
+    t.pass('ecstatic should be started')
+    checkServerIsRunning('http://0.0.0.0:' + process.env.PORT, ecstatic, t)
+  })
+})
+
+insanePorts.forEach(function (port) {
+  test('insane port: ' + port, function (t) {
+    t.plan(2)
+    process.env.PORT = port
```

Comment:
You can pass this into spawn as an argument---something like `{ env: { PORT: port }}` (we'd have to hit up the docs to be sure on that)